Dynamic display of temperatures for all CPU cores and NVMe drives.

eremem opened the pull request:
Hi,
I found your extension very useful. Since the predefined fixed number of cores and NVMe drives was too small for my setup, I extended your version, so that it shows the actual number of the mentioned items. This version works with PVE 7.4 and 8.0.

Meliox commented:
Hi,

That's awesome. Many have been asking for this. Thanks

I see that you add a temperature offset for core0? Why only that one - would this benefit all users?

eremem commented:
Hi, 
> I see that you add a temperature offset for core0? Why only that one - would this benefit all users?
I'm not sure if I understand correctly, its not an offset for the temperature but for the core sensor inputs. They start with that offset on my system and then the numbers raise continuously - temp2_input for core 0, temp3_input for core 1 etc.

Meliox commented:
I see, misunderstood it, thanks.

Can you move the configurable variable into the first section. That way is easier to configure for different cpu's, e.g. CPUtempInputOffset :)

eremem commented:
> Can you move the configurable variable into the first section. That way is easier to configure for different cpu's, e.g. CPUtempInputOffset :)

Sure, I'll adjust the code and push the changes soon.

> 
> I cannot get it to add the degree sign upon installation. It probably needs some escaping or might even be dependt on OS language.

That's strange, it should actually work everywhere because of the HTML entity. It's how it looks in my browser:

![](https://github.com/Meliox/PVE-mods/assets/56274896/66a5e969-945f-4797-8fc4-3fb476f83cb3)

What do you see in yours? Is only this character missing or the whole text?

Meliox commented:
Oh. Sorry. It does work. For some reason VS code didn't push update
through. All good now.

søn. 25. jun. 2023 19.25 skrev R.M.M ***@***.***>:

> Can you move the configurable variable into the first section. That way is
> easier to configure for different cpu's, e.g. CPUtempInputOffset :)
>
> Sure, I'll adjust the code and push the changes soon.
>
> I cannot get it to add the degree sign upon installation. It probably
> needs some escaping or might even be dependt on OS language.
>
> That's strange, it should actually work everywhere because of the HTML
> entity. It's how it looks in my browser:
>
>
> <https://user-images.githubusercontent.com/56274896/248574302-66a5e969-945f-4797-8fc4-3fb476f83cb3.png>
>
> What do you see in yours? Is only this character missing or the whole text?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Meliox/PVE-mods/pull/2#issuecomment-1606172279>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIFH4A2E7FCOVTL25BNG5DXNBX75ANCNFSM6AAAAAAZSXX52E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

eremem commented:
> Oh. Sorry. It does work. For some reason VS code didn't push update through. All good now.

No problem and good to hear it works:D

Please see the latest commit (https://github.com/Meliox/PVE-mods/pull/2/commits/77e06bccd2080f8c7b70848445c9f101c6c923f7) and feel free to rename the configuration constants.
I also added some comments and changed the declaration:
`let objValue = JSON.parse(value); `
to
`const objValue = JSON.parse(value);`
since objValue is not modified in the code, so the new declaration is more precise.

Meliox commented:
Just tested the new addition. Looks good. I pushed some changes tomove the configurable items to the beginning of the script. That will make it easier for the user to make changes :)
